content/tracing/setup/java.md

```diff
@@ -248,11 +248,105 @@ public class Application {
+
+Distributed Tracing headers are language agnostic. A trace started in one language may propagate to another (for example, from Python to Java).
+
+### Custom Distributed Tracing
+
+Create a distributed trace in custom instrumentation using OpenTracing:
+
+```java
+// Step 1: Inject the Datadog headers in the client code
+try (Scope scope = tracer.buildSpan("httpClientSpan").startActive(true)) {
+    final Span span = scope.span();
+    HttpRequest request = /* your code here */;
+
+    tracer.inject(span.context(),
+                  Format.Builtin.HTTP_HEADERS,
+                  new MyHttpHeadersInjectAdapter(request));
+
+    // http request impl...
+}
+
+public static class MyHttpHeadersInjectAdapter implements TextMap {
+  private final HttpRequest httpRequest;
+
+  public HttpHeadersInjectAdapter(final HttpRequest httpRequest) {
+    this.httpRequest = httpRequest;
+  }
+
+  @Override
+  public void put(final String key, final String value) {
+    httpRequest.addHeader(key, value);
+  }
+
+  @Override
+  public Iterator<Map.Entry<String, String>> iterator() {
+    throw new UnsupportedOperationException("This class should be used only with tracer#inject()");
+  }
+}
+
+// Step 2: Extract the Datadog headers in the server code
+HttpRequest request = /* your code here */;
+
+final SpanContext extractedContext =
+  GlobalTracer.get().extract(Format.Builtin.HTTP_HEADERS,
+                             new MyHttpRequestExtractAdapter(request));
+
+try (Scope scope = tracer.buildSpan("httpServerSpan").asChildOf(extractedContext).startActive(true)) {
+    final Span span = scope.span(); // will be a child of http client span in step 1
+    // http server impl...
+}
+
+public class MyHttpRequestExtractAdapter implements TextMap {
+  private final HttpRequest request;
+
+  public HttpRequestExtractAdapter(final HttpRequest request) {
+    this.request = request;
+  }
+
+  @Override
+  public Iterator<Map.Entry<String, String>> iterator() {
+    return request.headers().iterator();
+  }
+
+  @Override
+  public void put(final String key, final String value) {
+    throw new UnsupportedOperationException("This class should be used only with Tracer.extract()!");
+  }
+}
+```
 
-Enable priority sampling to ensure that distributed traces are complete.  Priority sampling achieves this by automatically assigning and propagating a priority value along all traces, depending on their service and volume.  Priorities can also be set manually to either drop non-interesting traces or to keep important ones.
+### Priority Sampling
+
+Distributed Traces may sampling inconsistently when the linked traces run on different hosts. Enable priority sampling to ensure that distributed traces are complete. Priority sampling achieves this by automatically assigning and propagating a priority value along all traces, depending on their service and volume. Priorities can also be set manually to either drop non-interesting traces or to keep important ones.
```

Comment:
Distributed Traces may sample inconsistently when the linked traces run on different hosts. To ensure that distributed traces are complete, enable priority sampling . Priority sampling automatically assigns and propagates a priority value along all traces, depending on their service and volume. Priorities can also be set manually to drop non-interesting traces or keep important ones.